markdown/markdown.go

```diff
@@ -266,11 +267,132 @@ func (r *renderer) RenderNode(ctx context.Context, w io.Writer, node *blackfrida
 				}
 				return blackfriday.GoToNext
 			}
+			if newNodes := rewriteDates(node); len(newNodes) > 0 {
+				for _, n := range newNodes {
+					if status := r.Renderer.RenderNode(w, n, entering); status != blackfriday.GoToNext {
+						return status
+					}
+				}
+				return blackfriday.GoToNext
+			}
 		}
 	}
 	return r.Renderer.RenderNode(w, node, entering)
 }
 
+var datePattern = regexp.MustCompile(
+	// For examples, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time#valid_datetime_values
+	// https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats
+	`\b(` +
+		// Month string: MM-DD
+		`[0-1][1-9]-[0-3][1-9]|` +
+		// Week string: YYYY-WXX
+		`\d{4}-W\d?\d|` +
+		// Date or datetime string, with optional timezone: YYYY-MM-DD HH:MM:SS+HH:MM
+		`\d{4}-\d{2}(-\d{2}([ T]?[0-2]?\d:\d\d(:\d\d(\.\d+)?)?(Z|[+-]\d\d:?\d\d)?)?)?|` +
+		// Time string, with optional timezone: HH:MM:SS+HH:MM+HH:MM
+		`(T?[0-2]?\d:\d\d(:\d\d(\.\d+)?)?(Z|[+-]\d\d:?\d\d)?)|` +
+		// Fiscal year with optional quarter
+		`FY[' -]?(\d{2,4})(?:[' -]?F?Q([1-4]))?|` +
+		// Fiscal quarter only
+		`F?Q([1-4])` +
+		`)\b`)
+
+var fiscalIntervalPattern = regexp.MustCompile(
+	`^(?:` +
+		// Fiscal year
+		`FY[' -]?(\d{2,4})(?:[' -]?F?Q([1-4]))?|` +
+		// Fiscal quarter only
+		`F?Q([1-4])` +
+		`)$`)
+
+// parseFiscalInterval returns a YYYY-MM-DD or MM-DD date string for a reference to a fiscal quarter or fiscal year,
+// for the start of the fiscal quarter or fiscal year, respectively.
+func parseFiscalInterval(dateStr string) string {
+	matches := fiscalIntervalPattern.FindStringSubmatch(dateStr)
+	if len(matches) == 0 {
+		return ""
+	}
+	fyStr := matches[1]
+	qStr := matches[2]
+	if fyStr == "" {
+		qStr = matches[3]
+	}
+
+	fq := 1
+	if qStr != "" {
+		var err error
+		fq, err = strconv.Atoi(qStr)
+		if err != nil {
+			panic(err)
```

Comment:
Instead of a panic, you should return an error. 

Looking above it seems on error we generally append an error to the renderer

Reply:
I acutally like the panic 🤷 It's just validating that the string that should only be digits (as checked by the regex pattern) is parseable as an int. If that invariant isn't true, something is wrong, and we should fail loudly.

Reply:
What @camdencheek said was my reasoning for using a panic here.